Feat: 커뮤니티 페이지 구현

yangseohyun opened the pull request:
## ✅ PR 유형
어떤 변경 사항이 있었나요?

- [x] 새로운 기능 추가
- [x] 버그 수정
- [ ] 리팩토링
- [ ] 코드에 영향을 주지 않는 변경사항(주석, 개행 등등..)
- [ ] 문서 수정
- [x] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 테스트 코드 추가

---

## ✏️ 작업 내용

![image](https://github.com/user-attachments/assets/13e9a80a-7dc0-4b88-b7a8-bfe6ef53506e)
-메인 화면
-검색할 경우, 모든 게시판을 대상으로 검색

![image](https://github.com/user-attachments/assets/b5f00fac-9e56-41ef-a712-3a4f940f7c60)
-커뮤니티 이동 기능

![image](https://github.com/user-attachments/assets/f137f34c-4206-4f06-a9a1-1189a76d4c8e)
-게시글 화면/댓글쓰기/답글쓰기
-"삭제하기/저장하기"는 본인이 작성한 글과 댓글/답글에서만 가능

---

## 🔗 관련 이슈
- closes #37

---

## 💡 추가 사항


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **신규 기능**
  - 커뮤니티 페이지에 팀별 게시글 목록, 상세 조회, 작성, 검색 기능이 추가되었습니다.
  - 커뮤니티 게시글의 댓글 및 답글 작성, 수정, 삭제 기능이 제공됩니다.
  - 인기 게시글, 파티 카드, 경기 일정 등 메인 페이지에 동적 컴포넌트가 도입되었습니다.
  - 검색 입력값을 전역에서 관리하는 기능이 추가되었습니다.
  - 팀 탭 네비게이션과 관련된 UI 컴포넌트가 추가되었습니다.
  - 게시글 목록 아이템과 팀 탭 네비게이션 UI가 새롭게 구현되었습니다.
  - 모달 컴포넌트에 키보드 접근성 및 포커스 트랩 기능이 강화되었습니다.
  - 라이브매치 다이어리 페이지에 챗봇 기능이 추가되었습니다.
  - 헤더에 검색 입력창이 추가되어 엔터키로 검색 실행 및 결과 페이지로 이동이 가능합니다.

- **스타일**
  - 커뮤니티, 메인, 게시글, 파티, 스케줄 등 신규 페이지 및 컴포넌트에 맞는 CSS 모듈이 대거 추가되었습니다.
  - 모달 스타일이 사용자 친화적으로 개선되고, 기존 스타일의 여백, 색상, 위치 등이 조정되었습니다.
  - 일부 기존 스타일의 여백, 색상, 위치 등이 개선되었습니다.

- **버그 수정**
  - 탭 네비게이션, 다이어리 등에서 불필요한 공백 및 속성 오타가 수정되었습니다.

- **리팩터**
  - 커뮤니티 관련 컴포넌트 구조가 개선되고, 일부 중복/불필요한 파일이 제거되었습니다.

- **기타**
  - 라우팅이 커뮤니티 게시글 작성, 상세, 검색 등 세부 페이지를 지원하도록 확장되었습니다.
  - 검색 기능 및 상태관리를 위한 컨텍스트가 도입되었습니다.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

yangseohyun committed:
proto_main

yangseohyun committed:
Feat: 커뮤니티 페이지 구현

zsjon committed:
Merge branch 'develop' into yangseohyun/issue37

zsjon commented on src/pages/Community/Community.module.css:
```diff
@@ -0,0 +1,282 @@
+  width: 100%;
+  justify-content: center;
+}
+.selectBtn {
```
selectBtn 클래스의 경우, 기타 모달과 같이 color를 #ff4f4f 로 수정해 주는 게 좋을 것 같습니다!

yangseohyun replied:
수정완료

zsjon commented on src/pages/Community/PostDetail.js:
```diff
@@ -0,0 +1,328 @@
+            <div className={styles.title}>{post.title}</div>
+            {isAuthor(post.author) && (
+              <div className={styles.menuWrapper}>
+                <button onClick={() => setShowPostMenu((prev) => !prev)} className={styles.menuButton}>수정/삭제</button>
```
해당 버튼은 수정/삭제 대신, '직관일지 작성' 부분에서 사용하는 대로 3점 메뉴로 디자인하는 것이 좋다고 판단됩니다. 그리고 가능하면 게시글 상세의 구조 자체가 댓글을 제외한 구조가 직관일지와 동일하게 가는 건 어떨지 의견이 궁금합니다!

zsjon commented on src/pages/Community/PostWrite.js:
해당 게시글 작성 로직에 ABS봇 비속어 감지 모달이 아직 적용되어 있지 않은 것 같습니다! 해당 부분은 Modal 컴포넌트를 활용해서 현재 develop 브랜치에 있는 Modal 컴포넌트 사용 예시를 보고 작성 부탁드립니다!
예시)
`
{showModal && (
                <Modal
                    title="ABS봇 작동중!"
                    message="ABS봇이 부적절한 키워드를 감지했습니다. 작성글을 수정해 주세요.
감지 내용: *#$&@*"
                    buttons={[
                        {label: '확인', onClick: () => {setShowModal(false)}
                        }
                    ]}
                />
            )}
            `

zsjon commented on src/pages/Community/SearchContext.js:
해당 코드 파일의 경우 검색 기능을 위한 상태 관리와 상태 공유를 간편하게 만들어주는 기능을 하므로, pages 디렉토리 대신 components 디렉토리에 위치하는 것이 좋다고 판단됩니다!

zsjon reviewed (changes requested):
comments 확인 부탁드립니다!

zsjon commented on src/pages/League/MatchInfo.js:
```diff
@@ -0,0 +1,223 @@
+        {id: '10', name: '안치홍', position: '2루수', hand: '우타'},
+        {id: '11', name: '김상복', position: '유격수', hand: '우타'},
+        {id: '12', name: '피경준', position: '3루수', hand: '좌타'}
+    ]);
```
해당 Issue는 PR merge 이후 재확인 예정입니다.

zsjon commented on src/components/TabNav/TabNav.js:
```diff
@@ -0,0 +1,57 @@
+        width: `${100 / tabCount}%`,
+        left: `${(100 / tabCount) * activeIndex}%`,
+    };
+
```
tabCount의 경우 PR merge 이후 적용할 예정입니다.

zsjon commented on src/components/TabNav/SubTabNav.module.css:
```diff
@@ -0,0 +1,52 @@
+  background-color: #0D1A80;
+  transition: left 0.3s ease;
+  border-radius: 2px;
+}
```
해당 Issue는 PR merge 이후 적용하도록 하겠습니다.

zsjon commented on src/components/TabNav/SubTabNav.module.css:
```diff
@@ -0,0 +1,52 @@
+  margin: 0 auto; /* 가운데 정렬 */
+  padding-bottom: 6px;
+  border-bottom: 1px solid #ccc;
+}
```
해당 디자인 개선 Issue의 경우 서비스가 제공되는 최소 조건의 작은 화면에서 확인 결과 이상 없음으로 판단하여 유지하겠습니다.

zsjon commented on src/components/TabNav/TabNav.module.css:
```diff
@@ -0,0 +1,52 @@
+  background-color: #0D1A80;
+  transition: left 0.3s ease;
+  border-radius: 2px;
+}
```
확인 결과 현재 의도에 맞게 적용되고 있는 것으로 확인되어 따로 수정을 적용하지 않겠습니다.

zsjon commented on src/components/DummyData/dummyDiaries.js:
```diff
@@ -0,0 +1,89 @@
+    team: '한화 이글스',
+    date: '2025.03.01',
+    image: `${process.env.PUBLIC_URL}/exImage2.png`,
+    teamLogo: `${process.env.PUBLIC_URL}/Logo/TeamLogo/emblem_HH.png`,
```
해당 Issue는 PR 이후 적용 예정입니다.

zsjon commented on src/components/TabNav/TabNav.module.css:
```diff
@@ -0,0 +1,52 @@
+  margin: 0 auto; /* 가운데 정렬 */
+  padding-bottom: 6px;
+  border-bottom: 1px solid #ccc;
+}
```
해당 Issue는 PR merge 이후 적용 예정입니다.

zsjon commented on src/pages/LiveMatchDiary/NewDiary.js:
```diff
@@ -0,0 +1,162 @@
+                    >
+                        {isEditing ? '수정 완료!' : '직관일지 작성!'}
+                    </button>
+                </div>
```
하지만 버튼 클릭 시 해당 form이 제출되도록 구현하려면 해당 button 클래스에 onClick을 줘야 하지 않나요?

zsjon replied:
알겠습니다. merge 이후에 수정 진행하도록 하겠습니다.

zsjon commented on src/pages/LiveMatchDiary/NewDiary.js:
```diff
@@ -0,0 +1,162 @@
+            setImage(URL.createObjectURL(e.target.files[0]));
+            setImageFile(e.target.files[0]);
+        }
+    };
```
해당 Issue는 PR merge 이후 확인해 보겠습니다.

zsjon commented on src/pages/LiveMatchDiary/DiaryDetail.js:
```diff
@@ -0,0 +1,99 @@
+                  }
+                ]}
+              />
+            )}
```
해당 Issue는 PR merge 이후 수정 예정입니다.

zsjon commented on src/pages/LiveMatchDiary/DiaryDetail.module.css:
```diff
@@ -0,0 +1,88 @@
+    justify-content: center;
+    width: 100%;
+    padding-bottom: 3.5rem;
+}
```
해당 Issue는 PR merge 이후 수정 예정입니다.

zsjon commented on src/pages/Profile/Profile.js:
```diff
@@ -0,0 +1,276 @@
+                              className={styles.editIcon}
+                              onClick={() => setShowProfileEditModal(true)}
+                          />
+                        )}
```
현재 develop 브랜치에서 Springboot API와 연동하면서 수정중입니다.

zsjon commented on src/pages/League/MatchInfo.js:
```diff
@@ -0,0 +1,223 @@
+                </div>
+            )}
+        </div>
+    );
```
이후 별도 컴포넌트로 분리 예정입니다.

yangseohyun commented on src/components/TabNav/TabNav.js:
```diff
@@ -0,0 +1,57 @@
+                        role="tab"
+                        id={`tab-${index}`}
+                        aria-selected={activeIndex === index}
+                        aria-controls={`tabpanel-${index}`}ㅡ
```
aria-controls={`tabpanel-${index}`}ㅡtabIndex={activeIndex === index ? 0 : -1} 이렇게가 한줄이야

yangseohyun commented on src/pages/Community/PostWrite.js:
```diff
@@ -0,0 +1,136 @@
+        }
+    }, [editingPost]);
+
+    const handleImageChange = (e) => {
+        const file = e.target.files[0];
+        if (file) {
+            setImageFile(file);
+            const url = URL.createObjectURL(file);
+            setObjectUrl(url);
+            setImage(url);
+        }
+    };
```
setImageFile을 사용하기 위해
// 이미지 파일이 있는 경우 FileReader를 사용하여 base64로 변환
        let imageData = image;
        if (imageFile) {
            const reader = new FileReader();
            imageData = await new Promise((resolve) => {
                reader.onloadend = () => {
                    resolve(reader.result);
                };
                reader.readAsDataURL(imageFile);
            });
        } 
코드를 추가했어

yangseohyun replied:
예

yangseohyun committed:
[Design] 디자인변경
-포스트작성화면 디자인 변경
직관일지와 통일
-검색결과없을때 화면 디자인 변경
검색결과있을때와의 통일감을 위해 css 변경
-게시글 상세화면 디자인 변경
수정/삭제 버튼 삼점바로 변경

[Refactor] 코드 추가 및 삭제
-Json 예외 방지
-중복된 코드 삭제

yangseohyun committed:
Merge branch 'yangseohyun/issue37'

yangseohyun commented on src/pages/Main/ScheduleSection.js:
```diff
@@ -0,0 +1,35 @@
+      </div>
+    </section>
+  );
+}
```
이건 Feat: 메인페이지 구현에서 할게

yangseohyun commented on src/pages/Main/MainPage.js:
```diff
@@ -1,29 +1,73 @@
+    schedule: { home: "NC", away: "삼성", status: "종료", score: [6, 3] },
+    posts: [
+      {
+        profile: "/profile/user2.jpg",
+        nickname: "NC팬1",
+        title: "오늘 경기 너무 재밌었어요!",
+        date: "2025.05.02 18:00",
+      },
+    ],
+    parties: [
+      {
+        image: "/profile/party1.png",
+        filters: ["승인제", "20대", "여자만"],
+        title: "NC 직관팟 모집",
+        author: "홍길동",
+        gender: "남성",
+        date: "5.5(일) 오후 2:00",
+        participants: 8,
+        maxParticipants: 12,
+      },
+    ],
+  },
+  "LG 트윈스": {
+    schedule: { home: "LG", away: "KT", status: "예정", score: [0, 0] },
+    posts: [
+      {
+        profile: "/profile/party1.png",
+        nickname: "엘지빠돌이",
+        title: "비 예보 있어서 걱정이네",
+        date: "2025.05.02 15:30",
+      },
+    ],
+    parties: [],
+  },
+  "삼성 라이온즈": {
+    schedule: { home: "삼성", away: "SSG", status: "경기중", score: [2, 4] },
+    posts: [],
+    parties: [],
+  },
+};
```
이것도 메인페이지 구현에서 할게.

yangseohyun commented on src/pages/Main/MainPage.js:
```diff
@@ -1,29 +1,73 @@
-                    <span>03.13(화)</span>
-                    <span>LG vs 두산 10 : 5</span>
-                </div>
-            </section>
+  const selectedTeam = "NC 다이노스"; // 기본값으로 NC 다이노스 선택
+  const { schedule, posts, parties } = sampleData[selectedTeam];
+
```
이것도 메인페이지구현에서 할게.

yangseohyun committed:
[Design] 디자인변경
 -포스트작성화면 디자인 변경 직관일지와 통일
 -검색결과없을때 화면 디자인 변경
  검색결과있을때와의 통일감을 위해 css 변경
 -게시글 상세화면 디자인 변경
 수정/삭제 버튼 삼점바로 변경

[Refactor]
-오류처리추가
-관리개선
-중복요소제거

yangseohyun committed:
[Fix]충돌 및 코드오류 수정

yangseohyun commented on src/pages/Community/Community.js:
```diff
@@ -0,0 +1,229 @@
+              className={styles.teamGrid}
+              role="radiogroup"
+              aria-label="팀 선택"
+            >
+              {teams.map((team) => (
+                <button
+                  key={team.id}
+                  className={
+                    tempTeam === team.id
+                      ? styles.teamBtnSelected
+                      : styles.teamBtn
+                  }
+                  onClick={() => setTempTeam(team.id)}
+                  role="radio"
+                  aria-checked={tempTeam === team.id}
+                  aria-label={`${team.name} 선택`}
+                >
+                  <img src={team.logo} alt="" className={styles.teamModalLogo} />
+                  <span>{team.name}</span>
+                </button>
+              ))}
+            </div>
+            <div className={styles.modalActions}>
+              <button 
+                className={styles.selectBtn} 
+                onClick={handleSelectTeam}
+                aria-label="선택한 팀으로 이동"
+              >
+                선택
+              </button>
+              <button 
+                className={styles.cancelBtn} 
+                onClick={handleCloseModal}
+                aria-label="팀 선택 취소"
+              >
+                취소
+              </button>
+            </div>
+          </div>
+        </div>
```
모달 모양이 달라서 재사용이 어려워.

zsjon reviewed (approved):
수고하셨습니다! 이슈 resolve 완료 확인했습니다.
